api/src/modules/characters/controllers/character.ctrl.spec.ts

```diff
@@ -0,0 +1,115 @@
+import assert from 'assert';
+
+import status from 'http-status-codes';
+
+import testApi from '../../../shared/spec/api';
+import { createTestUserAndLogin } from '../../../shared/spec/db';
+import {
+  createPostTestForGuardAgainstUnlogin,
+  createGetTestForGuardAgainstUnlogin,
+} from '../../../shared/spec/security';
+
+type createCharReqType = {
+  name: string;
+  race: string;
+};
+
+function createCharacter(
+  token: string,
+  creationPayload: createCharReqType = {
+    name: 'Some name',
+    race: 'elf',
+  }
+) {
+  return testApi(token)
+    .post('/users/current/characters')
+    .send(creationPayload);
+}
+
+describe('Character controller', () => {
+  describe('create new character', () => {
+    it('should not create character with not logged in user', async () => {
+      const createCharReq = {
+        name: 'Some name',
+        race: 'elf',
+      };
+
+      await createPostTestForGuardAgainstUnlogin(
+        '/users/current/characters',
+        createCharReq
+      );
+    });
+    it('should create character with logged in user', async () => {
+      const token = await createTestUserAndLogin();
+
+      const createCharReq: createCharReqType = {
+        name: 'Some name',
+        race: 'elf',
+      };
+
+      const { body } = await createCharacter(token, createCharReq).expect(
+        status.CREATED
+      );
+
+      assert.strictEqual(body.name, createCharReq.name);
+      assert.strictEqual(body.race.key, createCharReq.race);
+    });
+  });
+  describe('find one', () => {
+    it('should not find one with not logged in user', async () => {
+      const token = await createTestUserAndLogin();
+      const resCreation = await createCharacter(token);
+
+      await createGetTestForGuardAgainstUnlogin(
+        `/users/current/characters/${resCreation.body.id}`
+      );
+    });
+    it('created and found character should be same', async () => {
+      const token = await createTestUserAndLogin();
+      const resCreation = await createCharacter(token);
+
+      const oneCharacterResponse = await testApi(token)
+        .get(`/users/current/characters/${resCreation.body.id}`)
+        .expect(status.OK);
+
+      assert.deepStrictEqual(resCreation.body, oneCharacterResponse.body);
+    });
+
+    it('it send internal error on bad character id format', async () => {
+      const token = await createTestUserAndLogin();
+
+      await testApi(token)
+        .get(`/users/current/characters/ANY_NON_EX_ID`)
+        .expect(status.INTERNAL_SERVER_ERROR);
```

Comment:
it's clearly `badRequest`. 400 means that you provide data in wrong format etc.
INTERNAL SERVER ERROR should never happen in normal circumstances. 500 mean that something goes wrong and we not really know what, we did not handle it.
you should not test it as a normal case.

Reply:
additionally, in openapi you write that it can be any string.
but now it's throw strange errors when it is not uuid.